src/System.Linq.Expressions/tests/CompilerTests.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Xunit;
+
+namespace System.Linq.Expressions.Tests
+{
+    public static class CompilerTests
+    {
+        [Theory]
+        [ClassData(typeof(CompilationTypes))]
+        public static void CompileDeepTree(bool useInterpreter)
```

Comment:
Nit: for someone coming along later, consider making this CompileDeepTree_NoStackOverflow to highlight exactly what the test is verifying